Assets/HoloToolkit/SpatialSound/Scripts/AudioInfluencerManager.cs

```diff
@@ -0,0 +1,242 @@
+
+        // The collection of previously applied audio influencers.
+        private List<IAudioInfluencer> previousInfluencers = new List<IAudioInfluencer>();
+
+        // Potential effects manipulated by an audio influencer and their key
+        // properties.
+        private AudioLowPassFilter lowPassFilter;
+        private float nativeLowPassCutoffFrequency;
+        private AudioHighPassFilter highPassFilter;
+        private float nativeHighPassCutoffFrequency;
+
+        private void Awake() 
+        {           
+            audioSource = gameObject.GetComponent<AudioSource>();
+            initialAudioSourceVolume = audioSource.volume;
+
+            // Get optional filters (and initial values) that the sound designer / developer 
+            // may have applied to this game object
+            lowPassFilter = gameObject.GetComponent<AudioLowPassFilter>();
+            nativeLowPassCutoffFrequency = (lowPassFilter != null) ? lowPassFilter.cutoffFrequency : NeutralHighFrequency;
+            highPassFilter = gameObject.GetComponent<AudioHighPassFilter>();
+            nativeHighPassCutoffFrequency = (highPassFilter != null) ? highPassFilter.cutoffFrequency : NeutralLowFrequency;
+
+            // Preallocate the array that will be used to collect RaycastHit structures.
+            hits = new RaycastHit[MaxObjects];
+        }
+	
+	    private void Update() 
+        {
+            DateTime now = DateTime.Now;
+
+            // Audio influences are not updated every frame.
+            if ((UpdateInterval * 1000.0f) <= (now - lastUpdate).Milliseconds)
+            {
+                audioSource.volume = initialAudioSourceVolume;
+
+                // Get the audio influencers that should apply to the audio source.
+                List<IAudioInfluencer> influencers = GetInfluencers();
+                foreach (IAudioInfluencer influencer in influencers)
+                {
+                    // Apply the influencer's effect.
+                    influencer.ApplyEffect(gameObject);
+                }
+
+                // Find and remove the audio influencers that are to be removed from the audio source.
+                List<IAudioInfluencer> influencersToRemove = new List<IAudioInfluencer>();
+                foreach (IAudioInfluencer prev in previousInfluencers)
+                {
+                    MonoBehaviour mbPrev = prev as MonoBehaviour;
+
+                    // Remove influencers that are no longer in line of sight
+                    // OR
+                    // Have been disabled
+                    if (!influencers.Contains(prev) ||
+                        ((mbPrev != null) && !mbPrev.isActiveAndEnabled))
+                    {
+                        influencersToRemove.Add(prev);
+                    }
+                }
+                RemoveInfluencers(influencersToRemove);
+
+                previousInfluencers = influencers;
+                lastUpdate = now;
+            }
+	    }
```

Comment:
formatting seems off

Reply:
what part seems off? either my editor is set different from yours or I am just not seeing it :)

Reply:
spaces could be off.